### PR TITLE
Remove ParseError.partial

### PR DESCRIPTION
**Pull request by CAD97**
as it's just 0..at

### CODE REVIEW

File: src/generate/rust.rs [hunk 1]
```diff
@@ -852,7 +852,7 @@ where
                     input,
                     #code_label,
                     #parse_node_kind,
-                ).map(handle).map_err(|err| err.map_partial(|x| handle(x).source_info()))
+                ).map(handle)
```
Comment (eddyb) [inline src/generate/rust.rs]: Oh, can you inline the `handle` closure?

Comment (CAD97) [reply src/generate/rust.rs]: So
```rust
gll::runtime::Parser::parse(
    input,
    #code_label,
    #parse_node_kind,
).map(|forest_and_node| OwnedHandle {
    forest_and_node,
    _marker: PhantomData,
})
```
?

(This looks like a use case for a `OwnedHandle::new`, actually)

File: src/generate/templates/header.rs [hunk 2]
```diff
@@ -9,6 +9,13 @@ pub struct OwnedHandle<I: gll::runtime::Input, T: ?Sized> {
 }
 
 impl<I: gll::runtime::Input, T: ?Sized> OwnedHandle<I, T> {
+    pub fn new(forest_and_node: gll::runtime::OwnedParseForestAndNode<_P, I>) -> Self {
```
Comment (eddyb) [inline src/generate/templates/header.rs]: Bad idea, this is a type that should only be created by "privileged code", and be otherwise opaque externally.

Comment (CAD97) [reply src/generate/templates/header.rs]: Alright, the bad commit has been removed from history 💥

**Pull request merged**